### ISSUE TITLE
Flask_Ask 500 Internal error with Zappa

### ISSUE THREAD

**Issue by rschmidtz** [Miserlou/Zappa#933]
## Context

 

Hi, I followed this tutorial to deplot an Alexa Skill with Zappa: [video](https://www.youtube.com/watch?v=mjWV4R2P4ks) and I'm getting a 500 error

## Expected Behavior
My skill should work from Alexa(works locally with ngrok)

## Actual Behavior

Output from AWS API Gateway

`"{u'message': u'An uncaught exception happened while servicing this request. You can investigate this with the `zappa tail` command.', u'traceback': ['Traceback (most recent call last):\\n', '  File \"/var/task/handler.py\", line 433, in handler\\n    response = Response.from_app(self.wsgi_app, environ)\\n', '  File \"/tmp/pip-build-jW5QZX/Werkzeug/werkzeug/wrappers.py\", line 903, in from_app\\n', '  File \"/tmp/pip-build-jW5QZX/Werkzeug/werkzeug/wrappers.py\", line 57, in _run_wsgi_app\\n', '  File \"/tmp/pip-build-jW5QZX/Werkzeug/werkzeug/test.py\", line 884, in run_wsgi_app\\n', \"TypeError: 'NoneType' object is not callable\\n\"]}"`

## Steps to Reproduce

1. source venv/bin/activate
2. pip install flask_ask zappa
3. zappa init / zappa deploy dev

## Environment

* Zappa version used: zappa==0.42.1
* Operating System and Python version: Ubuntu / Python 2.7
* pip freeze`aniso8601==1.2.0
argcomplete==1.8.2
asn1crypto==0.22.0
base58==0.2.4
boto3==1.4.4
botocore==1.5.40
certifi==2017.4.17
cffi==1.10.0
chardet==3.0.4
click==6.7
cryptography==1.9
docutils==0.13.1
durationpy==0.4
enum34==1.1.6
Flask==0.12.1
Flask-Ask==0.9.3
future==0.16.0
futures==3.1.1
hjson==2.0.7
idna==2.5
ipaddress==1.0.18
itsdangerous==0.24
Jinja2==2.9.6
jmespath==0.9.3
kappa==0.6.0
lambda-packages==0.15.1
MarkupSafe==1.0
placebo==0.8.1
pycparser==2.17
pyOpenSSL==17.0.0
python-dateutil==2.6.0
python-slugify==1.2.4
PyYAML==3.12
requests==2.18.1
s3transfer==0.1.10
six==1.10.0
toml==0.9.2
tqdm==4.14.0
troposphere==1.9.4
Unidecode==0.4.20
urllib3==1.21.1
Werkzeug==0.12
wsgi-request-logger==0.4.6
zappa==0.42.1
`
* zappa_settings.json

```
{
    "dev": {
        "app_function": "skill.app", 
        "profile_name": "default", 
        "s3_bucket": "zapa-XXXX"
    }
}
```

Any thoughts?

**Comment by libert-xyz** [comment]
## Fix

setupstools problem, refer [link](https://github.com/pypa/setuptools/issues/937)

### Fix it with 

pip install packaging

**Comment by hmc-cs-abaugus** [comment]
Hey there,

I've got identical behavior from the same tutorial, but pip install packaging did not work for me; pip check finds no broken requirements, but zappa tail outputs the following: No module named 'cryptography.hazmat.bindings._openssl': ModuleNotFoundError. It seems like this may be a typo, as the previous line successfully does "from cryptography.hazmat.bindings.openssl.binding import Binding", but I don't know where to go to fix the typo (I think it's possible the part that says _openssl should just say openssl). I've uninstalled and reinstalled cryptography as well as reinstalled pip but neither fixed the issue. Has anyone else had this problem?

**Comment by yamat124** [comment]
I had this issue last night with my lambda. Got super scared but I solved by deleting all *.pyc __pycache__ folders and files. re-deploying my production and it came back up.

**Comment by robarthur** [comment]
Seeing a similar issue to the one listed above.
From my cloudwatch logs
`No module named packaging.version: ImportError`

Python 2.7 pip 9.0.1

`pip freeze
aniso8601==1.2.0
argcomplete==1.8.2
arrow==0.10.0
asn1crypto==0.22.0
base58==0.2.4
binaryornot==0.4.3
boto3==1.4.4
botocore==1.5.40
certifi==2017.4.17
cffi==1.10.0
chardet==3.0.4
click==6.7
cookiecutter==1.5.1
cryptography==1.9
docutils==0.13.1
durationpy==0.4
enum34==1.1.6
Flask==0.12.2
Flask-Ask==0.9.4
future==0.16.0
futures==3.1.1
hjson==2.0.7
idna==2.5
ipaddress==1.0.18
itsdangerous==0.24
Jinja2==2.9.6
jinja2-time==0.2.0
jmespath==0.9.3
kappa==0.6.0
lambda-packages==0.15.1
MarkupSafe==1.0
packaging==16.8
pkg-resources==0.0.0
placebo==0.8.1
poyo==0.4.1
pycparser==2.17
pyOpenSSL==17.0.0
pyparsing==2.2.0
python-dateutil==2.6.0
python-slugify==1.2.4
PyYAML==3.12
requests==2.18.1
s3transfer==0.1.10
six==1.10.0
toml==0.9.2
tqdm==4.14.0
troposphere==1.9.4
Unidecode==0.4.21
urllib3==1.21.1
Werkzeug==0.12.2
whichcraft==0.4.1
wsgi-request-logger==0.4.6
zappa==0.42.2
`
zappa_settings.json

`{
    "dev": {
        "app_function": "app.app",
        "aws_region": "eu-west-2",
        "s3_bucket": "zappa-xxxxxxxxx"
    }
}
`

Any suggestions what this could be?

**Comment by rschmidtz** [comment]
@robarthur installing packaging solved the problem in my case

`pip install packaging
`

**Comment by GeorgianaPetria** [comment]
I'm having the same problem and **pip install packaging** didn't work for me. Any other solution that worked?

**Comment by yamat124** [comment]
Deleting all Compile python files __pycache__ worked for me. Good luck

**Comment by GeorgianaPetria** [comment]
When do you delete the python files? 

I used removed all .pyc files in my virtual env `find . | grep -E "(__pycache__|\.pyc|\.pyo$)" | xargs rm -rf` and then `zappa update dev` but it still doesn't work.

**Comment by yamat124** [comment]
Does that do it recursively? I do it before deploying production.

**Comment by GeorgianaPetria** [comment]
Yes, it does it recursively, but unfortunately it's still not working?
Any idea how I can to narrow down the error? 
`"{u'message': u'An uncaught exception happened while servicing this request. You can investigate this with the `zappa tail` command.', u'traceback': ['Traceback (most recent call last):\\n', '  File \"/var/task/handler.py\", line 434, in handler\\n    response = Response.from_app(self.wsgi_app, environ)\\n', '  File \"/tmp/pip-build-0Sv96s/Werkzeug/werkzeug/wrappers.py\", line 903, in from_app\\n', '  File \"/tmp/pip-build-0Sv96s/Werkzeug/werkzeug/test.py\", line 884, in run_wsgi_app\\n', \"TypeError: 'NoneType' object is not callable\\n\"]}"`

**Comment by yamat124** [comment]
I started a new project from scratch and got this issue. Are you still having problems?

**Comment by RenegadeMaster** [comment]
I have successfully deployed several Zappa projects but only recently encountered this error. I also noticed for the first time, Zappa creation required cloudformation privileges. Could it be something in a newer version? This was a clean project, clean deploy

**Comment by variable** [comment]
Having the same problem with Django...

**Comment by Arttii** [comment]
I am having the same problem with Apistar, its complaining about dependencies, that are there.
